pymeasure/instruments/rigol/rigol_ds1000z.py

```diff
@@ -0,0 +1,870 @@
+        target, numeric = _normalize_memory_depth_value(value)
+        if numeric is not None:
+            allowed = self._allowed_memory_depth_values()
+            if numeric not in allowed:
+                raise ValueError(
+                    f"Memory depth {numeric} pts is not permitted for the current "
+                    f"channel configuration. Allowed values: {sorted(allowed)} or 'AUTO'."
+                )
+        self.write(f":ACQ:MDEP {target}")
+
+    trigger_source = Instrument.control(
+        ":TRIG:EDGE:SOUR?",
+        ":TRIG:EDGE:SOUR %s",
+        "Control the trigger source selection (str).",
+        validator=strict_discrete_set,
+        values=_WAVEFORM_SOURCES,
+    )
+
+    trigger_slope = Instrument.control(
+        ":TRIG:EDGE:SLOP?",
+        ":TRIG:EDGE:SLOP %s",
+        "Control the trigger slope (str): 'POS' or 'NEG'.",
+        validator=strict_discrete_set,
+        values=["POS", "NEG"],
+    )
+
+    trigger_coupling = Instrument.control(
+        ":TRIG:COUP?",
+        ":TRIG:COUP %s",
+        "Control the trigger coupling mode (str).",
+        validator=strict_discrete_set,
+        values=["AC", "DC", "HFRej", "LFRej", "NOIS"],
+    )
+
+    trigger_sweep = Instrument.control(
+        ":TRIG:SWE?",
+        ":TRIG:SWE %s",
+        "Control the trigger sweep mode (str): 'AUTO', 'NORM', or 'SING'.",
+        validator=strict_discrete_set,
+        values=["AUTO", "NORM", "SING"],
+    )
+
+    trigger_level = Instrument.control(
+        ":TRIG:LEV?",
+        ":TRIG:LEV %f",
+        "Control the trigger level in volts (float).",
+        cast=float,
+    )
+
+    waveform_source = Instrument.control(
+        ":WAV:SOUR?",
+        ":WAV:SOUR %s",
+        "Control the waveform source used for data transfers (str).",
+        validator=strict_discrete_set,
+        values=_WAVEFORM_SOURCES,
+    )
+
+    waveform_mode = Instrument.control(
+        ":WAV:MODE?",
+        ":WAV:MODE %s",
+        "Control the waveform acquisition mode for transfer (str).",
+        validator=strict_discrete_set,
+        values=[mode.value for mode in WaveformMode],
+    )
+
+    waveform_format = Instrument.control(
+        ":WAV:FORM?",
+        ":WAV:FORM %s",
+        "Control the waveform data format for transfer (str).",
+        validator=strict_discrete_set,
+        values=[fmt.value for fmt in WaveformFormat],
+    )
+
+    waveform_start = Instrument.control(
+        ":WAV:STAR?",
+        ":WAV:STAR %d",
+        "Control the first data point index for waveform transfer (int).",
+        cast=int,
+    )
+
+    waveform_stop = Instrument.control(
+        ":WAV:STOP?",
+        ":WAV:STOP %d",
+        "Control the last data point index for waveform transfer (int).",
+        cast=int,
+    )
+
+    def __init__(self, adapter, name="Rigol DS1000Z/MSO1000Z Oscilloscope",
+                 channel_count=4, **kwargs):
+        """Initialise the driver.
```

Comment:
An init should not have a docstring. Instead, the docstring (and parameters) should go to the class itself.
That way, if you write `inst = Rigol...(adapter...` you will get the docstring of the instrument and will see the parameters. The __init__, you basically never call directly.